xvec/accessor.py

```diff
@@ -683,6 +692,195 @@ def set_geom_indexes(
         _obj = _obj.drop_indexes(coord_names)
 
         for coord in coord_names:
+            if crs:
+                _obj[coord].attrs["crs"] = CRS.from_user_input(crs)
             _obj = _obj.set_xindex(coord, GeometryIndex, crs=crs, **kwargs)
 
         return _obj
+
+    def to_geopandas(self):
+        """Convert this array into a GeoPandas :class:`~geopandas.GeoDataFrame`
+
+        Returns a :class:`~geopandas.GeoDataFrame` with coordinates based on a
+        :class:`~xvec.GeometryIndex` set as an active geometry. The array needs to
+        contain only a single geometry-based coordinate axis and can be 1D or 2D. Other
+        cases are not supported.
```

Comment:
Would it make sense to add an argument for specifying the coordinate name to use as active geometry?

Reply:
Yeah, I did that in `to_geodataframe` and forgot to add the same here.

Reply:
Actually, it would not make sense. The maximum dimensionality that can be coerced to a DataFrame using `to_pandas` is 2D. In that case, if an index is geometry, we reset it and set as an active geom. If columns are geometry, we transpose the DataFrame, reset index and set geometry as active. If both index and columns are geometry, we raise an error (l.750) as we cannot keep a DataFrame indexed by geometry since GeoPandas does not support that use case. So there's no case where a keyword would change anything.